### PR TITLE
Allow embedders to add events to the timeline.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Fixes https://github.com/flutter/flutter/issues/28283

### CODE REVIEW

File: shell/platform/embedder/embedder.h [hunk 1]
```diff
@@ -592,6 +592,28 @@ FlutterEngineResult FlutterEngineDispatchSemanticsAction(
     const uint8_t* data,
     size_t data_length);
 
+// A profiling utility. Logs a trace duration begin event to the timeline. If
+// the timeline is not available or disabled, this has no effect. Must be
```
Comment (jason-simmons) [inline shell/platform/embedder/embedder.h]: Change "not available" to "unavailable" in these comments to reduce ambiguity

File: shell/platform/embedder/embedder.h [hunk 2]
```diff
@@ -592,6 +592,28 @@ FlutterEngineResult FlutterEngineDispatchSemanticsAction(
...
+// balanced with an duration end event (via
+// |FlutterEngineTraceEventDurationEnd|) with the same name on the same thread.
+// Can be called on any thread.
+FLUTTER_EXPORT
+void FlutterEngineTraceEventDurationBegin(const char* name);
+
+// A profiling utility. Logs a trace duration end event to the timeline. If
+// the timeline is not available or disabled, this has no effect. This call must
+// be preceeded by a trace duration begin call (via
+// FlutterEngineTraceEventDurationBegin) with the same name on the same thread.
```
Comment (liyuqian) [inline shell/platform/embedder/embedder.h]: Nit: `|FlutterEngineTraceEventDurationBegin|`

File: shell/platform/embedder/embedder.h (hunk 1 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done.

File: shell/platform/embedder/embedder.h (hunk 2 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done.

**Pull request merged**